# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../dovecot/dovecot-2.2-0006-854469baa57c.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---


# HG changeset patch
# User Timo Sirainen <tss@iki.fi>
# Date 1366118430 -10800
# Node ID 854469baa57c7e3b32fdcf71928048e7db95e291
# Parent  f33bacb03cc22e02ca2d11d527d8dc0d99214486
auth: Fixed multiple master passdbs.

diff -r f33bacb03cc2 -r 854469baa57c src/auth/auth-request.c
--- a/src/auth/auth-request.c	Tue Apr 16 15:54:05 2013 +0300
+++ b/src/auth/auth-request.c	Tue Apr 16 16:20:30 2013 +0300
@@ -583,7 +583,8 @@
 		request->skip_password_check = TRUE;
 	}
 
-	if (request->requested_login_user != NULL) {
+	if (request->requested_login_user != NULL &&
+	    *result == PASSDB_RESULT_OK) {
 		auth_request_master_lookup_finish(request);
 		/* if the passdb lookup continues, it continues with non-master
 		   passdbs for the requested_login_user. */

